File: HeroscapeBuilder.Server/Domain/Entities/PointSystem.cs
using System.Text.Json.Serialization;

namespace HeroscapeBuilder.Server.Domain.Entities
{
    /// <summary>
    /// The competing sets of unit point values. Standard is the baseline every unit has;
    /// Renegade and Delta override it for some units.
    /// </summary>
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum PointSystem
    {
        /// <summary>Hasbro's original points, or the printed points for units Renegade produced.</summary>
        Standard,

        /// <summary>Renegade's official points adjustments to original units.</summary>
        Renegade,

        /// <summary>Community-maintained Delta points.</summary>
        Delta,
    }

    public static class PointSystemExtensions
    {
        public const PointSystem Default = PointSystem.Renegade;

        /// <summary>
        /// The unit's points under <paramref name="system"/>, falling back to Standard when there is no override.
        /// </summary>
        public static int? Resolve(this PointSystem system, int? standard, int? renegade, int? delta)
        {
            var overridePoints = system switch
            {
                PointSystem.Renegade => renegade,
                PointSystem.Delta => delta,
                _ => null,
            };
            return overridePoints ?? standard;
        }

        public static int? PointsFor(this UnitEntity unit, PointSystem system)
        {
            return system.Resolve(unit.StandardPoints, unit.RenegadePoints, unit.DeltaPoints);
        }
    }
}

File: heroscapebuilder.client/src/models/point-system.ts
import { Unit } from './unit';

/**
 * The competing sets of unit point values. Every unit has Standard points; Renegade and Delta
 * override them for some units and fall back to Standard otherwise.
 */
export type PointSystem = 'Standard' | 'Renegade' | 'Delta';

export const DEFAULT_POINT_SYSTEM: PointSystem = 'Renegade';

export interface PointSystemInfo {
    value: PointSystem;
    label: string;
    description: string;
}

export const POINT_SYSTEMS: PointSystemInfo[] = [
    {
        value: 'Renegade',
        label: 'Renegade',
        description: "Renegade's official points, including its adjustments to original Hasbro units.",
    },
    {
        value: 'Standard',
        label: 'Standard',
        description: 'The original points printed by Hasbro (and by Renegade for its own units), with no later adjustments.',
    },
    {
        value: 'Delta',
        label: 'Delta',
        description: 'Community-maintained Delta points from heroscape.org. Units without a Delta value use Standard.',
    },
];

export const isPointSystem = (value: unknown): value is PointSystem =>
    POINT_SYSTEMS.some(system => system.value === value);

type PointValues = Pick<Unit, 'standardPoints' | 'renegadePoints' | 'deltaPoints'>;

/** The unit's points under the given system, falling back to Standard when it has no override. */
export const getUnitPoints = (unit: PointValues | undefined | null, system: PointSystem): number | undefined => {
    if (!unit) return undefined;
    const override = system === 'Renegade' ? unit.renegadePoints
        : system === 'Delta' ? unit.deltaPoints
        : undefined;
    return override ?? unit.standardPoints ?? undefined;
};
